netbox/ipam/api/serializers.py

```diff
@@ -33,6 +34,10 @@ class Meta:
             'custom_fields', 'created', 'last_updated', 'asn_count',
         ]
 
+    @extend_schema_field(OpenApiTypes.INT)
+    def get_asn_count(self, obj):
+        return obj.get_child_asns().count()
+
```

Comment:
There are two issues with this approach:

1. It generates a separate query per object in the table.
2. It prevents us from ordering the table by ASN count.

I've opened #13053 with a more complete (albeit more complex) solution. Please let me know what you think.